src/anyvar/restapi/main.py

```diff
@@ -0,0 +1,235 @@
+    :raise HTTPException: if requested location isn't found
+    """
+    av: AnyVar = request.app.state.anyvar
+    try:
+        location = av.get_object(location_id)
+    except KeyError:
+        return HTTPException(status_code=404)
+    return {"location": location.as_dict()}
+
+
+@app.put(
+    "/variation",
+    response_model=RegisterVariationResponse,
+    summary="Register a new variation object",
+    description="Provide a variation definition to be normalized and registered with AnyVar. A complete VRS object and digest is returned for later reference.",  # noqa: E501
+    tags=[EndpointTag.VARIATIONS]
+)
+def register_variation(
+    request: Request,
+    variation: RegisterVariationRequest = Body(
+        description="Variation description, including (at minimum) a definition property"  # noqa: E501
+    )
+):
+    """Register a variation based on a provided description or reference.
+
+    :param request: FastAPI request object
+    :param variation: provided variation description
+    :return: messages describing translation failure, or object and references if
+        successful
+    """
+    av: AnyVar = request.app.state.anyvar
+    definition = variation.definition
+    result = {
+        "object": None,
+        "messages": []
+    }
+    try:
+        translated_variation = av.translator.translate(
+            var=definition,
+            untranslatable_to_text=variation.untranslatable_to_text
+        )
+    except TranslationException:
+        result["messages"].append(f"Unable to translate {definition}")
+    except NotImplementedError:
+        result["messages"].append(
+            f"Variation class for {definition} is currently unsupported."
+        )
+    else:
+        v_id = av.put_object(translated_variation)
+        result["object"] = translated_variation.as_dict()
+        result["object_id"] = v_id
+    return result
+
+
+@app.get(
+    "/variation/{variation_id}",
+    response_model=GetVariationResponse,
+    operation_id="getVariation",
+    summary="Retrieve a variation object",
+    description="Gets a variation instance by ID. May return any supported type of variation.",  # noqa: E501
+    tags=[EndpointTag.VARIATIONS]
+)
+def get_variation_by_id(
+    request: Request,
+    variation_id: StrictStr = Query(..., description="VRS ID for variation")
+):
+    """Get registered variation given VRS ID.
+
+    :param request: FastAPI request object
+    :param variation_id: ID to look up
+    :return: VRS variation if successful
+    :raise HTTPException: if no variation matches provided ID
+    """
+    av: AnyVar = request.app.state.anyvar
+
+    try:
+        variation = av.get_object(variation_id, deref=True)
+    except KeyError:
+        raise HTTPException(
+            status_code=404, detail=f"Variation {variation_id} not found"
+        )
+
+    result = {"messages": [], "data": variation.as_dict()}
+
+    return result
+
+
+@app.get(
+    "/search",
```

Comment:
I couldn't get genomic to work using `search?accession=ga4gh%3ASQ.F-LrLMe1SRpfUZHkQmvkVKFEGaoDeHul&start=140753336&end=140753336` after creating the genomic variation. I know we worked on this a long time ago, so not sure what's happening. 

I tried using the protein sequence and got an internal server error:

```
INFO:     127.0.0.1:55763 - "GET /search?accession=ga4gh%3ASQ.cQvw4UsHHRRlogxbWCB8W-mKD4AraM9y&start=599&end=600 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/fastapi/routing.py", line 165, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "biocommons/anyvar/src/anyvar/restapi/main.py", line 197, in search_variations
    inline_alleles.append(av.get_object(allele["_id"], deref=True).asdict())
  File "biocommons/anyvar/venv/lib/python3.10/site-packages/python_jsonschema_objects/classbuilder.py", line 265, in __getattr__
    raise AttributeError(
AttributeError: asdict is not a valid property of Allele
```